schema/signature.go

```diff
@@ -25,3 +27,34 @@ type FunctionSignature struct {
 	// parameter if it is supported.
 	VarParam *function.Parameter
 }
+
+func (fs *FunctionSignature) Copy() *FunctionSignature {
+	newFS := &FunctionSignature{
+		Description: fs.Description,
+		Detail:      fs.Detail,
+		ReturnType:  fs.ReturnType,
+		VarParam:    copyFunctionParameter(fs.VarParam),
```

Comment:
Instead of using a helper function here, that reimplements some of the `cty` internals (field names) and would need to be updated whenever something upstream changes, we can make a copy using some pointer magic.

```go
if fs.VarParam != nil {
	vpCpy := *fs.VarParam
	newFS.VarParam = &vpCpy
}
```

Reply:
Ah, neat! And that works because there are no nested pointers, right? Or would that be irrelevant anyway?

Reply:
While it possible to have nested pointers in Go (`**T`), I don't think it's relevant here.